sub_peers.js

```diff
@@ -35,6 +35,7 @@ function TChannelSubPeers(channel, options) {
     var self = this;
     this.peerScoreThreshold = this.options.peerScoreThreshold || 0;
     this.choosePeerWithHeap = channel.choosePeerWithHeap;
+    this.socketInitTimeout = channel.initTimeout;
```

Comment:
Because `Peer` and `Connection` class are managed by `rootChannel`.

The best way to apply sub channel options is in the `SubPeers` class, utilizing mutation.